File: employee-user-api/supermarketVue/src/model/entity/SupermarketSale.ts
// in js, the class object would auto inherit serializable, so it do not need to implement serializable like java
class SupermarketSale {
    id: number | null;
    branch: string | null;
    city: string | null;
    customerType: string | null;
    gender: string | null;
    productLine: string | null;
    unitPrice: number | null;
    quantity: number | null;
    tax5Percent: number | null;
    total: number | null;
    date: Date | null;
    time: Date | null;
    payment: string | null;
    cogs: number | null;
    grossMarginPercentage: number | null;
    grossIncome: number | null;
    rating: number | null;

    constructor(
        id: number| null,
        branch: string| null,
        city: string| null,
        customerType: string| null,
        gender: string| null,
        productLine: string| null,
        unitPrice: number| null,
        quantity: number| null,
        tax5Percent: number| null,
        total: number| null,
        date: Date| null,
        time: Date| null,
        payment: string| null,
        cogs: number| null,
        grossMarginPercentage: number| null,
        grossIncome: number| null,
        rating: number| null
    ) {
        this.id = id;
        this.branch = branch;
        this.city = city;
        this.customerType = customerType;
        this.gender = gender;
        this.productLine = productLine;
        this.unitPrice = unitPrice;
        this.quantity = quantity;
        this.tax5Percent = tax5Percent;
        this.total = total;
        this.date = date;
        this.time = time;
        this.payment = payment;
        this.cogs = cogs;
        this.grossMarginPercentage = grossMarginPercentage;
        this.grossIncome = grossIncome;
        this.rating = rating;
    }
}

export default SupermarketSale;

File: sideProject/Springboot/model/src/main/java/org/zane/SupermarketSale.java
package org.zane;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SupermarketSale implements Serializable {
    private Long id;

    private String branch;

    private String city;

    private String customerType;

    private String gender;

    private String productLine;

    private BigDecimal unitPrice;

    private int quantity;

    private BigDecimal tax5Percent;

    private BigDecimal total;

    private Date date;

    private Date time;

    private String payment;

    private BigDecimal cogs;

    private BigDecimal grossMarginPercentage;

    private BigDecimal grossIncome;

    private BigDecimal rating;
}


